pallets/loan/src/tests.rs

```diff
@@ -853,5 +855,483 @@ fn test_pool_nav() {
+		.build()
+		.execute_with(|| {
+			let pool_account = pallet_pool::Pallet::<MockRuntime>::account_id();
+			let owner: u64 = 10;
+			let pool_balance = balance_of::<MockRuntime, GetUSDCurrencyId>(&pool_account);
+			assert_eq!(pool_balance, 1000 * USD);
+
+			let owner_balance = balance_of::<MockRuntime, GetUSDCurrencyId>(&owner);
+			assert_eq!(owner_balance, Zero::zero());
+
+			let pool_id = create_pool::<MockRuntime, GetUSDCurrencyId>(owner);
+			let asset_registry = create_nft_registry::<MockRuntime>(owner);
+			let token_id = mint_nft::<MockRuntime>(owner, asset_registry);
+			let res = Loan::issue_loan(
+				Origin::signed(owner),
+				pool_id,
+				AssetId(asset_registry, token_id),
+			);
+			assert_ok!(res);
+
+			let loan_event = fetch_loan_event(last_event()).expect("should be a loan event");
+			let (pool_id, loan_id) = match loan_event {
+				LoanEvent::LoanIssued(pool_id, loan_id) => Some((pool_id, loan_id)),
+				_ => None,
+			}
+			.expect("must be a Loan issue event");
+
+			// activate loan
+			// interest rate is 5%
+			let rate = math::rate_per_sec(Rate::saturating_from_rational(5, 100)).unwrap();
+			let oracle: u64 = 1;
+			let loan_type = LoanType::BulletLoan(BulletLoan::new(
+				// advance rate 80%
+				Rate::saturating_from_rational(80, 100),
+				// expected loss over asset maturity 0.15%
+				Rate::saturating_from_rational(15, 10000),
+				// collateral value
+				Amount::from_inner(125 * USD),
+				// 4%
+				math::rate_per_sec(Rate::saturating_from_rational(4, 100)).unwrap(),
+				// maturity in 2 years
+				math::seconds_per_year() * 2,
+			));
+			let res = Loan::activate_loan(
+				Origin::signed(oracle),
+				pool_id,
+				loan_id,
+				rate,
+				// ceiling is 100 USD
+				loan_type,
+			);
+			assert_ok!(res);
+
+			// borrow 50
+			Timestamp::set_timestamp(1);
+			let borrow_amount = Amount::from_inner(50 * USD);
+			let res = Loan::borrow(Origin::signed(owner), pool_id, loan_id, borrow_amount);
+			assert_ok!(res);
+
+			// after one year
+			// caller should be admin, can write off before maturity
+			let caller = 1;
+			Timestamp::set_timestamp(math::seconds_per_year());
+			let res = Loan::admin_write_off_loan(Origin::signed(caller), pool_id, loan_id, 0);
+			assert_err!(res, Error::<MockRuntime>::ErrInvalidWriteOffGroupIndex);
+
+			// let the maturity date passes + 1 day
+			let t = math::seconds_per_year() * 2 + math::seconds_per_day();
+			Timestamp::set_timestamp(t);
+			let res = Loan::admin_write_off_loan(Origin::signed(caller), pool_id, loan_id, 0);
+			assert_err!(res, Error::<MockRuntime>::ErrInvalidWriteOffGroupIndex);
+
+			// add write off groups
+			for group in vec![(3, 10), (5, 15), (7, 20), (20, 30)] {
+				let res = Loan::add_write_off_group(
+					Origin::signed(oracle),
+					pool_id,
+					WriteOffGroup {
+						percentage: Rate::saturating_from_rational(group.1, 100),
+						overdue_days: group.0,
+					},
+				);
+				assert_ok!(res);
+			}
+
+			// index
```

Comment:
Could we make this comment more descriptive please

Reply:
sure